lib/rubocop/cop/style/class_structure.rb

```diff
@@ -0,0 +1,312 @@
+              add_offense(node, location: :expression, message: msg)
+            end
+            previous = index
+          end
+        end
+
+        # Autocorrect by swapping between two nodes autocorrecting them
+        def autocorrect(node)
+          node_classification = classify(node)
+          previous = left_siblings_of(node).find do |sibling|
+            classification = classify(sibling)
+            !ignore?(classification) && node_classification != classification
+          end
+
+          current_range = source_range_with_comment(node)
+          previous_range = source_range_with_comment(previous)
+
+          lambda do |corrector|
+            corrector.insert_before(previous_range, current_range.source)
+            corrector.remove(current_range)
+          end
+        end
+
+        private
+
+        # Classifies a node to match with something in the {expected_order}
+        # @param node to be analysed
+        # @return String when the node type is a `:block` then
+        #   {classify} recursively with the first children
+        # @return String when the node type is a `:send` then {find_category}
+        #   by method name
+        # @return String otherwise trying to {humanize_node} of the current node
+        def classify(node)
+          return node.to_s unless node.respond_to?(:type)
+          case node.type
+          when :block
+            classify(node.children.first)
+          when :send
+            find_category(node.method_name)
+          else
+            humanize_node(node)
+          end.to_s
+        end
+
+        # Categorize a method_name according to the {expected_order}
+        # @param method_name try to match {categories} values
+        # @return [String] with the key category or the `method_name` as string
+        def find_category(method_name)
+          name = method_name.to_s
+          categories.each do |category, method_names|
+            return category if method_names.include?(name)
+          end
+          name
+        end
+
+        def walk_over_nested_class_definition(class_node)
+          class_elements(class_node).each do |node|
+            classification = classify(node)
+            next if ignore?(classification)
+            yield node, classification
+          end
+        end
+
+        def class_elements(class_node)
+          *, class_def = class_node.children
+          return [] unless class_def
+          if class_def.def_type? || class_def.send_type?
+            [class_def]
+          else
+            class_def.children.compact
+          end
+        end
+
+        def ignore?(classification)
+          classification.nil? ||
+            classification.to_s.end_with?('=') ||
+            expected_order.index(classification).nil?
+        end
+
+        def node_visibility(node)
+          _, method_name, = *find_visibility_start(node)
+          method_name || :public
+        end
+
+        def find_visibility_start(node)
```

Comment:
There's common code that resolves visibility scope elsewhere in the codebase. Please see if you can use that instead of re-implementing. 🙂

Reply:
Great! I'll check if I find it.